src/cocotb/types/logic_array.py

```diff
@@ -320,10 +339,23 @@ def _get_str(self) -> str:
                 )
         return self._value_as_str
 
-    def _get_int(self) -> int:
+    def _get_int(
+        self,
+        resolve: "Literal['error'] | Literal['low'] | Literal['high'] | Literal['random']",
+    ) -> int:
         if self._value_as_int is None:
             # May convert list to str before converting to int.
-            self._value_as_int = int(self._get_str(), 2)
+            try:
+                self._value_as_int = int(self._get_str(), 2)
+            except ValueError:
+                if resolve == "error":
+                    raise
+
+                return int(
+                    RE_UNRESOLVED.sub(RESOLVE_MAP[resolve], self._get_str()),
```

Comment:
Apart from typing, a check for valid values of `resolve` would be nice. Or just use an enum (maybe even preferred).

Reply:
I prefer to not force people to import something extra just to pass an argument, which is why I'm of a fan of the literal type in place of enums, and not a fan of how Range() was added to LogicArray. I think StrEnum is a good combination, but is only available in 3.11. I can add a backport though.